src/main/java/edu/ucsb/mapache/controllers/SlackSlashCommandController.java

```diff
@@ -0,0 +1,259 @@
+import org.springframework.web.bind.annotation.RestController;
+
+import edu.ucsb.mapache.models.SlackSlashCommandParams;
+import edu.ucsb.mapache.repositories.ChannelRepository;
+
+// imports for google search
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.util.ArrayList;
+//
+
+
+/**
+ * Sample Slash Command Handler.
+ *
+ * @author ramswaroop
+ * @version 1.0.0, 20/06/2016
+ */
+@RestController
+public class SlackSlashCommandController {
+
+    private static final Logger logger = LoggerFactory.getLogger(SlackSlashCommandController.class);
+
+    @Autowired
+    ChannelRepository channelRepository;
+
+    /**
+     * The token you get while creating a new Slash Command. You should paste the
+     * token in application.properties file.
+     */
+    @Value("${app.slack.slashCommandToken}")
+    private String slackToken;
+
+    public String getSlackToken() {
+        return slackToken;
+    }
+
+    /**
+     * Slash Command handler. When a user types for example "/app help" then slack
+     * sends a POST request to this endpoint. So, this endpoint should match the url
+     * you set while creating the Slack Slash Command.
+     *
+     * @param token
+     * @param teamId
+     * @param teamDomain
+     * @param channelId
+     * @param channelName
+     * @param userId
+     * @param userName
+     * @param command
+     * @param text
+     * @param responseUrl
+     * @return
+     */
+    @RequestMapping(value = "/api/public/slash-command", method = RequestMethod.POST, consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
+    public RichMessage onReceiveSlashCommand(@RequestParam("token") String token,
+            @RequestParam("team_id") String teamId, @RequestParam("team_domain") String teamDomain,
+            @RequestParam("channel_id") String channelId, @RequestParam("channel_name") String channelName,
+            @RequestParam("user_id") String userId, @RequestParam("user_name") String userName,
+            @RequestParam("command") String command, @RequestParam("text") String text,
+            @RequestParam("response_url") String responseUrl) {
+        // validate token
+        logger.info("slash command processing...");
+        if (!token.equals(slackToken)) {
+            return new RichMessage("Sorry: the slack bot received an invalid token.");
+        }
+
+        SlackSlashCommandParams params = new SlackSlashCommandParams();
+
+        params.setToken(token);
+        params.setTeamId(teamId);
+        params.setTeamDomain(teamDomain);
+        params.setChannelId(channelId);
+        params.setChannelName(channelName);
+        params.setUserId(userId);
+        params.setCommand(command);
+        params.setText(text);
+        params.setResponseUrl(responseUrl);
+
+        // String textParts[] = params.getTextParts(); // <------- btkh4 advice was enacted here
```

Comment:
It isn't necessary to keep old code or to have comments mentioning that you changed something in response to a code review.

Reply:
Done, deleted the comments